### ISSUE TITLE
Cannot read property 'baseUrl' of undefined

### ISSUE THREAD

**Issue by hrz11** [formio/ngFormio#411]
I am getting this error:

angular.js:14642 TypeError: Cannot read property 'baseUrl' of undefined
    at new 228.module.exports.app.config.formioComponentsProvider.register.controller (formio.js:16948)
    at Object.instantiate (angular.js:5055)
    at $controller (angular.js:10975)
    at new 240.module.exports.formioComponents.controller (formio.js:18596)
    at Object.invoke (angular.js:5040)
    at $controllerInit (angular.js:10960)
    at nodeLinkFn (angular.js:9840)
    at angular.js:10248
    at processQueue (angular.js:17000)
    at angular.js:17044

code itself on the line 16946 is
     ```
case 'resource':
                var url = '';
                var baseUrl = $scope.options.baseUrl || Formio.getBaseUrl();
                //var baseUrl = Formio.getBaseUrl();
```

$scope.options is undefined
Looking around the code, it has been resolved in many places with code 
`$scope.options = $scope.options || {};`

ng-formio v2.20.3

**Comment by travist** [comment]
Thanks. I am releasing a new version within the hour that will fix this problem.

**Comment by travist** [comment]
Should be fixed with v2.20.4. Thanks.

**Comment by hrz11** [comment]
Thanks!!!

but....
angular.js:14642 TypeError: Cannot read property 'baseUrl' of undefined
    at new controller (formio.js:15713)
    at Object.invoke (angular.js:5040)
    at $controllerInit (angular.js:10960)
    at nodeLinkFn (angular.js:9840)
    at compositeLinkFn (angular.js:9149)
    at compositeLinkFn (angular.js:9152)
    at publicLinkFn (angular.js:9014)
    at lazyCompilation (angular.js:9405)
    at boundTranscludeFn (angular.js:9192)
    at controllersBoundTransclude (angular.js:9959)

different line now 15713 :)
there may be more of those...
